substrate/frame/staking-async/runtimes/parachain/src/weights/pallet_staking_async.rs

```diff
@@ -1,80 +1,103 @@
-// Copyright (C) Parity Technologies (UK) Ltd.
 // This file is part of Substrate.
 
-// Substrate is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
 
-// Substrate is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
-// You should have received a copy of the GNU General Public License
-// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
 //! Autogenerated weights for `pallet_staking_async`
 //!
+//! This is a special template for `election-provider-multi-block` pallet. This is required because
+// ! we don't want to generate the `trait WeightInfo`.
+//!
 //! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 32.0.0
-//! DATE: 2025-05-08, STEPS: `50`, REPEAT: `20`, LOW RANGE: `[]`, HIGH RANGE: `[]`
+//! DATE: 2025-06-30, STEPS: `20`, REPEAT: `10`, LOW RANGE: `[]`, HIGH RANGE: `[]`
 //! WORST CASE MAP SIZE: `1000000`
-//! HOSTNAME: `M3-Box`, CPU: `<UNKNOWN>`
-//! WASM-EXECUTION: `Compiled`, CHAIN: `None`, DB CACHE: 1024
+//! HOSTNAME: `ggwpez-ref-hw`, CPU: `AMD EPYC 7232P 8-Core Processor`
+//! WASM-EXECUTION: `Compiled`, CHAIN: `None`, DB CACHE: `1024`
 
 // Executed Command:
-// frame-omni-bencher
+// ../../../../../target/release/frame-omni-bencher
 // v1
 // benchmark
 // pallet
-// --runtime
-// target/release/wbuild/pallet-staking-async-parachain-runtime/pallet_staking_async_parachain_runtime.compact.compressed.wasm
 // --pallet
 // pallet_staking_async
 // --extrinsic
-//
+// *
+// --runtime
+// ../../../../../target/release/wbuild/pallet-staking-async-parachain-runtime/pallet_staking_async_parachain_runtime.compact.wasm
+// --steps
+// 20
+// --repeat
+// 10
+// --genesis-builder-preset
+// dot_size
+// --template
+// ../../../../../substrate/frame/election-provider-multi-block/src/template.hbs
```

Comment:
Dumb question: Why are we using a custom template? Also, is it intentional to use it for staking async pallet as well?

Reply:
actually no, this is just a artifact of me using the same script as I used for epmb pallets. 

The template is almost the same though, and compatible with staking async as well. 

Since this is a testing runtime, I will discount if for now and let it be.